src/components/Api.js

```diff
@@ -0,0 +1,140 @@
+export default class Api {
+  constructor(options) {
+    this._options = options;
+  }
+  _getUserAvatar() {
+    return fetch(this._options.baseUrl + "/users/me/avatar", {
+      headers: {
+        authorization: this._options.headers.authorization,
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+  _getUserData() {
+    return fetch(`${this._options.baseUrl}/users/me`, {
+      headers: {
+        authorization: this._options.headers.authorization,
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+  _getInitialCards() {
+    return fetch(`${this._options.baseUrl}/cards`, {
+      headers: {
+        authorization: this._options.headers.authorization,
+      },
+    }).then((res) => {
+      if (res.ok) {
+        return res.json();
+      }
+      return Promise.reject(`Error: ${res.status}`);
+    });
+  }
+  returnData() {
+    return Promise.all([this._getUserData(), this._getInitialCards()]);
+  }
+  patchUserInfo({ name, about }) {
+    return fetch(`${this._options.baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: {
+        authorization: this._options.headers.authorization,
+        "Content-Type": "application/json",
+      },
```

Comment:
According to the task you need to pass full `headers` into the constructor of `Api`. See here https://snipboard.io/pRdaAZ.jpg . And in the code you need to use only: `headers: this._headers` (not to duplicate the code)